sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/util/ValidationUtil.java

```diff
@@ -101,6 +101,18 @@ public static void validateTenantIdCharacterRange(String id, ClientLogger logger
         }
     }
 
+    public static void validateSubscriptionCharacterRange(String subscription, ClientLogger logger) {
+        if (subscription != null) {
+            for (int i = 0; i < subscription.length(); i++) {
+                if (!isValidSubscriptionCharacter(subscription.charAt(i))) {
+                    throw logger.logExceptionAsError(new IllegalArgumentException(
+                        "Invalid subscription id provided. You can locate your subscription id by following the instructions"
```

Comment:
```suggestion
                        "Invalid subscription ID provided. You can locate your subscription ID by following the instructions"
```

Reply:
Our validation rejects valid subscription names, so this error should suggest using an ID in that case. See https://github.com/Azure/azure-sdk-for-go/pull/23976